cylc/flow/task_pool.py

```diff
@@ -288,7 +288,14 @@ def compute_runahead(self):
                         TASK_STATUS_SUCCEEDED,
                         TASK_STATUS_EXPIRED
                     )
```

Comment:
Is this test still relevant (other than in back-compat mode)?
Any task that has failed or succeeded will only be in the pool if it is incomplete?
Expired tasks immediately get removed from the pool?

Reply:
I think you're right, but it's off-topic for this change, and now that you've raised it I want to re-check that expired tasks work as expected.  #4971